chromium_src/third_party/blink/renderer/modules/storage/dom_window_storage.cc

```diff
@@ -223,8 +223,11 @@ StorageArea* BraveDOMWindowStorage::ephemeralLocalStorage() {
   if (!namespaces)
     return nullptr;
 
-  auto storage_area =
-      namespaces->local_storage()->GetCachedArea(window->GetSecurityOrigin());
+  auto* controller = StorageController::GetInstance();
+  controller->ClearAreasIfNeeded();
```

Comment:
Comparing this fix with the code in `StorageNamespace::GetCachedArea()` I understand that this is basically a way of not trusting the cached storage area returned by the local storage for a given security domain when a new EphemeralStorage needs to be created for the DOM Window... by making sure that we always create a new cached area instead for such origin.

The only thing that I wonder whether it's needed is this call to `ClearAreasIfNeeded()`: I understand it's done for parity with [what `StorageNamespace::GetCachedAread()` does before creating a new area](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/storage/storage_namespace.cc;l=99?q=GetCachedArea&ss=chromium), but that call involves some checks + [a `for` loop](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/storage/storage_controller.cc;l=116;drc=a1748f2b248498bb0a48de23e1e0faff07fb805f) which, for each iteration, calls `CleanUpUnusedAreas()` which involves [a second `for` loop as well](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/storage/storage_namespace.cc;drc=a1748f2b248498bb0a48de23e1e0faff07fb805f;l=156), to actually remove the areas... and I wonder if that's strictly necessary for this fix. And if it's not, I think it would perhaps be better to leave it out of it, WDYT?

Reply:
I think it's still relevant to deal with storage quotas. This isn't technically a "new" storage area, it's a new connection to an existing storage area shared across tabs for a given domain. Mainly what we're trying to do here is to avoid incorrect local caching of values that have been deleted from the browser side memory store